### PR TITLE
Rewrite the library to use mio and tokio

### PR DESCRIPTION
**Pull request by eeeeeta**
This is still partially WIP (doesn't have any docs), but it works (and has been working perfectly for me for almost half a year now :P).

### CODE REVIEW

**Comment by zslayton** [comment]
This is GREAT! Thanks so much for putting the PR together! I'll set aside some time in the next couple of days to dust off my STOMP knowledge and review it.

File: Cargo.toml [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 description = "A full STOMP 1.2 client implementation. Allows programs to interact with message queueing services like ActiveMQ and RabbitMQ."
-keywords = [ "stomp", "mq", "activemq", "rabbitmq", "queue" ] 
+keywords = ["stomp", "mq", "activemq", "rabbitmq", "queue"]
 license = "MIT"
-
-[lib]
 name = "stomp"
+readme = "README.md"
+repository = "https://github.com/zslayton/stomp-rs"
+version = "0.13.1"
```
Comment (zslayton) [inline Cargo.toml]: Since the current (broken) version of `stomp-rs` is v0.11.0, could you change this to v0.12.0?

File: src/frame.rs [hunk 2]
```diff
@@ -1,52 +1,100 @@
...
-use std::io::Write;
-use std::io::Read;
-use std::io::BufRead;
 use std::str::from_utf8;
 use std::fmt;
 use std::fmt::Formatter;
+use bytes::BytesMut;
 
+#[derive(Copy, Clone, Debug)]
+pub enum StompCommand {
```
Comment (zslayton) [inline src/frame.rs]: Since this lives in `stomp::frame`, could the enum be named simply `Command`?

File: Cargo.toml (hunk 1 above)
Comment (eeeeeta) [reply Cargo.toml]: Fixed.

**Comment by eeeeeta** [comment]
Oh, wait, hang on, I committed an extra thing somehow, sec...

**Comment by eeeeeta** [comment]
There we are, should be good now.

**Comment by eeeeeta** [comment]
@zslayton Nits addressed and squash merge done (after a not-insignificant amount of fiddling with git...!)

**Comment by zslayton** [comment]
Merged, thanks very much! I noticed that one or two of the examples in the README needs to be updated now (e.g. `if frame.header.command == "CONNECTED"` is no longer valid). I'll cut a new cargo version once I've had a chance to look for any other updates I need to make.

**Pull request merged**